jiant/models.py

```diff
@@ -160,16 +161,18 @@ def build_sent_encoder(args, vocab, d_emb, tasks, embedder, cove_layer):
         log.info("Using PRPN sentence encoder!")
     elif any(isinstance(task, LanguageModelingTask) for task in tasks) or args.sent_enc == "bilm":
         assert_for_log(args.sent_enc in ["rnn", "bilm"], "Only RNNLM supported!")
-        assert_for_log(
-            not (
-                args.input_module == "elmo"
-                or args.input_module.startswith("bert")
-                or args.input_module.startswith("xlnet")
-            ),
-            f"Using input_module = {args.input_module} for language modeling is probably not a "
-            "good idea, since it allows the language model to use information from the right-hand "
-            "context.",
-        )
+        if not any(isinstance(task, MaskedLanguageModelingTask) for task in tasks):
+            # If an autoregressive LanguageModelingTask
```

Comment:
I'm not sure this is crucial, but it'd be nice to avoid this weird comment logic by explicitly introducing an AutoregressiveLMTask class.